src/transformers/models/van/modeling_van.py

```diff
@@ -0,0 +1,588 @@
+        patch_size: int,
+        stride: int,
+        depth: int,
+        mlp_expansion: int = 4,
+        drop_path_rate: float = 0.0,
+        dropout_rate: float = 0.0,
+        hidden_act: str = "gelu",
+        layer_norm_eps: float = 1e-6,
+        layer_scale_init_value: float = 1e-2,
+    ):
+        super().__init__()
+        self.embeddings = VanOverlappingPatchEmbedder(in_channels, hidden_size, patch_size, stride)
+        self.layers = nn.Sequential(
+            *[
+                VanLayer(
+                    hidden_size,
+                    mlp_expansion=mlp_expansion,
+                    drop_path_rate=drop_path_rate,
+                    dropout_rate=dropout_rate,
+                    hidden_act=hidden_act,
+                    layer_scale_init_value=layer_scale_init_value,
+                )
+                for _ in range(depth)
+            ]
+        )
+        self.norm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
+
+    def forward(self, hidden_state):
+        hidden_state = self.embeddings(hidden_state)
+        hidden_state = self.layers(hidden_state)
+        # rearrange b c h w -> b (h w) c
+        batch_size, hidden_size, height, width = hidden_state.shape
+        hidden_state = hidden_state.flatten(2).transpose(1, 2)
+        hidden_state = self.norm(hidden_state)
+        # rearrange  b (h w) c- > b c h w
+        hidden_state = hidden_state.view(batch_size, height, width, hidden_size).permute(0, 3, 1, 2)
+        return hidden_state
+
+
+class VanEncoder(nn.Module):
+    """
+    VanEncoder, consisting of multiple stages.
+    """
+
+    def __init__(self, config: VanConfig):
+        super().__init__()
+        self.stages = nn.ModuleList([])
+        patch_sizes = config.patch_sizes
+        strides = config.strides
+        hidden_sizes = config.hidden_sizes
+        depths = config.depths
+        mlp_expansions = config.mlp_expansions
+        drop_path_rates = [x.item() for x in torch.linspace(0, config.drop_path_rate, sum(config.depths))]
+
+        for num_stage, (patch_size, stride, hidden_size, depth, mlp_expantion, drop_path_rate) in enumerate(
+            zip(patch_sizes, strides, hidden_sizes, depths, mlp_expansions, drop_path_rates)
+        ):
+            is_first_stage = num_stage == 0
+            in_channels = hidden_sizes[num_stage - 1]
+            if is_first_stage:
+                in_channels = config.num_channels
+            self.stages.append(
+                VanStage(
+                    in_channels,
+                    hidden_size,
+                    patch_size=patch_size,
+                    stride=stride,
+                    depth=depth,
+                    mlp_expansion=mlp_expantion,
+                    drop_path_rate=drop_path_rate,
+                    dropout_rate=config.dropout_rate,
+                    layer_norm_eps=config.layer_norm_eps,
+                    layer_scale_init_value=config.layer_scale_init_value,
+                )
```

Comment:
Would be easier to just pass the config here.

Reply:
Maybe you can just do `config.hidden_sizes[num_stage]` etc.